src/core/iomgr/iomgr.c

```diff
@@ -74,19 +75,25 @@ static void background_callback_executor(void *ignored) {
       gpr_mu_lock(&g_mu);
     } else if (grpc_alarm_check(&g_mu, gpr_now(), &deadline)) {
     } else {
-      gpr_cv_wait(&g_cv, &g_mu, deadline);
+      gpr_mu_unlock(&g_mu);
+      gpr_sleep_until(gpr_time_min(short_deadline, deadline));
+      gpr_mu_lock(&g_mu);
     }
   }
   gpr_mu_unlock(&g_mu);
   gpr_event_set(&g_background_callback_executor_done, (void *)1);
 }
 
-void grpc_kick_poller(void) { gpr_cv_broadcast(&g_cv); }
+void grpc_kick_poller(void) {
```

Comment:
Maybe, maybe not. This is the path that alarms can use to get something in
the system running earlier. I suspect we won't need it, but I think it's
worth keeping this machinery alive for O(months) yet. Losing it makes it
hard to put back :)

On Tue, May 12, 2015 at 4:20 PM Nicolas Noble <notifications@github.com>
wrote:

> In src/core/iomgr/iomgr.c
> <https://github.com/grpc/grpc/pull/1567#discussion_r30190605>:
>
> >      }
> >    }
> >    gpr_mu_unlock(&g_mu);
> >    gpr_event_set(&g_background_callback_executor_done, (void *)1);
> >  }
> >
> > -void grpc_kick_poller(void) { gpr_cv_broadcast(&g_cv); }
> > +void grpc_kick_poller(void) {
>
> Is the kick mechanism at all relevant anymore at that point ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/1567/files#r30190605>.
>